config/dynamic.c

```diff
@@ -514,17 +512,33 @@ run_dynamic_config(struct gk_config *gk_conf, struct gt_config *gt_conf,
 			"dyn_cfg: Failed to unlink(%s) - (%s)\n",
 			dy_conf->server_path, strerror(errno));
 		ret = -1;
-		goto free_path;
+		goto free_server_path;
 	}
 
+	dy_conf->lua_dy_base_dir = rte_malloc(
+		"lua_dy_base_dir", strlen(lua_dy_base_dir) + 1, 0);
+	if (dy_conf->lua_dy_base_dir == NULL) {
+		ret = -1;
+		goto free_server_path;
+	}
+	strcpy(dy_conf->lua_dy_base_dir, lua_dy_base_dir);
+
+	dy_conf->dynamic_config_file = rte_malloc(
+		"dynamic_config_file", strlen(dynamic_config_file) + 1, 0);
```

Comment:
1. Apply `rte_strdup()` here as well.

2. You forgot to update `cleanup_dy()` for `dy_conf->lua_dy_base_dir` and `dy_conf->dynamic_config_file`.